### ISSUE TITLE
Table first/last child padding updates

### ISSUE THREAD

**Issue by mattnolting** [patternfly/patternfly#7032]
TLDR;

Table `:first/:last-child` css declarations are relics that should be standardized. As per a recent conversation w/ @lboehling, `:first/:last-child` `th/td` padding definitions can/should be simplified, while inset updates should be optional.

Why this is important:
- `:first/:last-child` are detached `:pseudo` elements and are calculated by the browser on every page
- These selectors cause a `force-reflow`, which significantly impacts LCP ([Largest Contentful Paint](https://developer.chrome.com/docs/lighthouse/performance/lighthouse-largest-contentful-paint/?utm_source=lighthouse&utm_medium=devtools)), which is the time the largest content element in the viewport is rendered to the screen.
- It's unnecessary 

This update includes
- [ ] Remove default `:first/:last-child` padding modifications
- [ ] Add optional `pf-m-inset-[sm, md, lg, xl]` `:first/:last-child` padding modifier

[Backstop report](https://drive.google.com/file/d/1dimvgl3HCcT0TIZC9olhuACTWPpvNA3z/view?usp=sharing)

**Comment by mattnolting** [comment]
The modifier can be added as an enhancement and does not block v6 release. 

@lboehling Can you confirm the modifier request and in set sizing value?

**Comment by mcoker** [comment]
closed by https://github.com/patternfly/patternfly/pull/7125
